rocky/views/ooi_list.py

```diff
@@ -24,9 +43,94 @@ def get_context_data(self, **kwargs):
 
         context["types_display"] = self.get_ooi_types_display()
         context["object_type_filters"] = self.get_ooi_type_filters()
+        context["select_oois_form"] = SelectOOIForm(context.get("ooi_list", []))
+        context["scan_levels"] = [alias for level, alias in SCAN_LEVEL.choices]
 
         return context
 
+    def get(self, request: HttpRequest, status=200, *args, **kwargs) -> HttpResponse:
+        context = self.get_context_data(**kwargs)
+        return self.render_to_response(context, status=status)
+
+    def post(self, request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        """Perform bulk action on selected oois."""
+        selected_oois = request.POST.getlist("ooi")
+
+        if not selected_oois:
+            messages.add_message(request, messages.ERROR, _("No OOIs selected."))
+            return self.get(request, status=422, *args, **kwargs)
+
+        action = request.POST.get("action")
+
+        if action == PageActions.DELETE.value:
+            return self._delete_oois(selected_oois, request, *args, **kwargs)
+
+        if action == PageActions.UPDATE_SCAN_PROFILE.value:
+            return self._set_scan_profiles(selected_oois, request, *args, **kwargs)
+
+        messages.add_message(request, messages.ERROR, _("Unknown action."))
+        return self.get(request, status=404, *args, **kwargs)
+
+    def _set_scan_profiles(self, selected_oois: List[Reference], request: HttpRequest, *args, **kwargs) -> HttpResponse:
+        connector = self.get_api_connector()
+        scan_profile = request.POST.get("scan-profile")
+
+        for level, alias in SCAN_LEVEL.choices:
```

Comment:
I'm not a huge fan of this loop, even though its just a few loops at worst.

Reply:
I am more worried about the one at line 82, but Octopoes does not support batched operations at this point for saving scan profiles and deleting objects (see https://github.com/minvws/nl-kat-octopoes/issues/28).

Or was this meant as a stylistic comment, in which case I can refactor?

Reply:
It was meant as a scalability improvement, looping over a set to find the right one seems silly, but is not too much of  a problem with only 4 items in that list.